### PR TITLE
Add the check of empty GET parameter

### LINKED ISSUE #26: taskcluster_login.py: Can't login to taskcluster, ~/tc_credentials.json is empty
Same root cause as #22: do_GET() is called twice. Now that #22 landed, we end up with tc_credentials.json which contains just `{}`.

This bug is not occasional.  Each time I'd like to login, I trigger this bug 2 or 3 times, before being actually connected.

For more details, [see this comment](https://github.com/askeing/taskcluster-util-python/issues/22#issuecomment-161583461)

### PR DESCRIPTION
**Pull request by askeing**
fix #26

### CODE REVIEW

**Comment by askeing** [comment]
ref: #24 

@JohanLorenzo @shinglyu r?

File: taskcluster_util/taskcluster_login.py [hunk 1]
```diff
@@ -115,20 +115,23 @@ def do_GET(self):
...
-            self.send_header('Content-type', 'text/html')
-            self.end_headers()
-            self.wfile.write('<h1>Login Successful</h1><br>You can close this window now...')
-            # stop server
-            logger.debug('======= Stop Server =======')
-            assassin = threading.Thread(target=self.server.shutdown)
-            assassin.daemon = True
-            assassin.start()
+            if 'certificate' not in query_components:
+                logger.warning('Do not get certificate from GET Parameters.'.format(query_components))
```
Comment (JohanLorenzo) [inline taskcluster_util/taskcluster_login.py]: Nit: 'Did not get

File: taskcluster_util/taskcluster_login.py [hunk 2]
```diff
@@ -115,20 +115,23 @@ def do_GET(self):
...
+            else:
+                # write to file
+                logger.info('Write credentials to {}'.format(credentials_file))
```
Comment (JohanLorenzo) [inline taskcluster_util/taskcluster_login.py]: I guess we can make `do_GET()` less complex by splitting these 13 lines into 3:

``` py
else:
    self.credentials_file_handler.save(content=query_components)
    self.respond_successful_login()
    self.stop()
```

Then you define these functions like:

``` py
class ServerHandler(BaseHTTPRequestHandler):
    # ...
    def respond_successful_login(self):
           self.send_response(200)
           self.send_header('Content-type', 'text/html')
           self.end_headers()
           self.wfile.write('<h1>Login Successful</h1><br>You can close this window now...')

    def stop(self):
        logger.debug('======= Stop Server =======')
        assassin = threading.Thread(target=self.server.shutdown)
        assassin.daemon = True
        assassin.start()
```

``` py
class CredentialsFileHandler:
    def __init__(self, file_path=None):
        self.file_path = file_path if file_path is not None else os.path.join(os.path.expanduser('~'), 'tc_credentials.json')

    def save(self, content):
        with open(self.file_path, mode='w') as f:
            json.dump(content, f, indent=4)

    def remove_exist_file(self):
        # put the content of _remove_exist_file() in here
```

What do you think?

**Comment by askeing** [comment]
@JohanLorenzo r? again :P

File: taskcluster_util/taskcluster_login.py [hunk 3]
```diff
@@ -90,6 +90,23 @@ def cli(self):
         return self
 
     class ServerHandler(BaseHTTPRequestHandler):
+        def credentials_file_handler(self, content):
```
Comment (JohanLorenzo) [inline taskcluster_util/taskcluster_login.py]: Nit: if the file hanlder is not a class, I'd put a verb in this function name. For example: `write_to_crendentials_file()`

**Comment by JohanLorenzo** [comment]
r+ modulo the small nit https://github.com/askeing/taskcluster-util-python/pull/29/files#r49171429

**Pull request merged**